### DISCUSSION TITLE
How to create a list within a list?

### DISCUSSION THREAD

**Discussion by therealchrismcfly** [neuefische/questions#169] [Web] [answered]
Hello Fischis,

here are some information to help understand what I'm about to do:

- there is a list where you can add your DIY projects
- when you click the edit button of a project, you can add some information to every project.

Link to the project: https://diy-project-planner.vercel.app/
Link to repo (useStore): https://github.com/therealchrismcfly/diy-project-planner/blob/main/src/hooks/useStore.js
Link to repo (edit Page): https://github.com/therealchrismcfly/diy-project-planner/blob/main/src/pages/edit/%5Bid%5D.js

**My problem:**
When I write down a note within a project, it appears on every project instead of just the one I added the note to. I know that the notes should somehow be attached to the project, but how?

<img width="746" alt="Bildschirmfoto 2022-09-23 um 17 03 36" src="https://user-images.githubusercontent.com/107792731/192100887-beda7795-6ae2-4129-a132-b566fc59148e.png">
<img width="730" alt="Bildschirmfoto 2022-09-23 um 17 03 53" src="https://user-images.githubusercontent.com/107792731/192100892-b6c7f3db-7d6e-4d93-afef-b89134c70e9e.png">
<img width="686" alt="Bildschirmfoto 2022-09-23 um 17 04 06" src="https://user-images.githubusercontent.com/107792731/192100893-b874939c-5f4d-4db9-8a2e-1bf9685c1880.png">

Thanks a lot for your help!

**Comment by quargh** [comment]
Hey Chris,
 
I would put the note inside of the projects object like this:
`{name: 'Pflanzenleiter', id: nanoid(), isDone: false, note:'some note'}`
and then edit / delete a note the same way you would switch isDone to true

Stefan

**Reply by therealchrismcfly** [reply]
Hey Stefan,

danke für deine Antwort. Aber wie würdest du addNote im useStore dann schreiben? 

Liebe Grüße
Chris

**Comment by quargh** [comment]
Hey Chris,

also... anders als ich es vorher geschrieben hatte, willst du ja einen Array und keinen einfachen String.
Insofern, im useStore:

`{name: 'Pflanzenleiter', id: nanoid(), isDone: false, notes: []},`

Die Funktion addNote:

`
	addNote: (id, note) => {
		set(state => ({
			projects: state.projects.map(project =>
				project.id === id ? {...project, notes: [...project.notes, note]} : project
			),
		}));
	},
`

In deiner edit Page, onSubmit:

`addNote(entry.id, note);`

In der edit Page unten die notes hinschreiben:

`
	{entry.notes.map((note, index) => {
		return <p key={index}>{note}</p>;
	})}
`

Erfolgreich geklont und getestet ;-)

LG, 
Stefan

**Reply by therealchrismcfly** [reply]
Vielen vielen Dank lieber Stefan! Es funktioniert jetzt bei den existierenden Datensätzen, leider aber noch nicht bei denen, die neu generiert werden, da kommt die folgende Fehlermeldung: 

<img width="955" alt="Bildschirmfoto 2022-09-29 um 08 48 39" src="https://user-images.githubusercontent.com/107792731/192959325-240d7afb-3cd1-4ef0-8640-361fd386245a.png">

Hier der Link zum Branch: 
https://github.com/therealchrismcfly/diy-project-planner/pull/20

Wirklich danke! Ich hänge jetzt seit einer Woche dran und nicht mal auf stackoverflow gabs eine Antwort!

**Comment by quargh** [comment, marked as answer]
Hello Chris,

sehr gerne :-)
Zu dem Problem mit dem neuen Datensatz, da müsstest du auch den notes-Array anlegen so wie bei den existierenden Datensätzen

Also in addProject:
`projects: [...state.projects, {id: nanoid(), name, isDone: false, notes: []}],`

LG Stefan

**Reply by therealchrismcfly** [reply]
Aaaah, das hatte ich, aber ohne [] und es funktioniert nun! 🥳 Vielen vielen vielen Dank!
